lib/modules/manager/composer/artifacts.ts

```diff
@@ -55,7 +55,11 @@ function getAuthJson(): string | null {
   }
 
   hostRules.findAll({ hostType: 'gitlab' })?.forEach((gitlabHostRule) => {
-    if (gitlabHostRule?.token) {
+    if (
+      (!gitlabHostRule.artifactAuth ||
+        gitlabHostRule.artifactAuth.includes('composer')) &&
+      gitlabHostRule?.token
+    ) {
```

Comment:
This is too narrowly applying to just gitlab tokens, should apply to all hostRules used in this file

Reply:
Should this also be applied for github?